daml-foundations/daml-ghc/ghc-compiler/src/DA/Daml/GHC/Compiler/Convert.hs

```diff
@@ -1107,36 +1101,39 @@ convertCast env expr co = do
 -- Coercions induced by newtypes can also occur deeply nested in function
 -- types or forall quantifications. We handle those cases by recursion into
 -- all sub-coercions.
-convertCoercion :: Env -> Coercion -> StateT Int ConvertM (LF.Expr -> LF.Expr, LF.Expr -> LF.Expr)
-convertCoercion env co@(coercionKind -> Pair s t)
-    | isReflCo co = pure (id, id)
-    | Just (aCo, bCo) <- splitFunCo_maybe co = do
-        let Pair a a' = coercionKind aCo
-        (aTo, aFrom) <- convertCoercion env aCo
-        (bTo, bFrom) <- convertCoercion env bCo
-        a <- lift $ convertType env a
-        a' <- lift $ convertType env a'
-        x <- mkLamBinder
-        let to expr = ETmLam (x, a') $ bTo $ ETmApp expr $ aFrom $ EVar x
-        let from expr = ETmLam (x, a) $ bFrom $ ETmApp expr $ aTo $ EVar x
-        pure (to, from)
-    -- NOTE(MH): This case is commented out because we don't know how to trigger
-    -- it in a test case yet. In theory it should do the right thing though.
-    -- | Just (a, k_co, co') <- splitForAllCo_maybe co
-    -- , isReflCo k_co
-    -- = do
-    --     (a, k) <- lift $ convTypeVar a
-    --     (to', from') <- convertCoercion env co'
-    --     let to expr = ETyLam (a, k) $ to' $ ETyApp expr $ TVar a
-    --     let from expr = ETyLam (a, k) $ from' $ ETyApp expr $ TVar a
-    --     pure (to, from)
-    -- Case (1) & (2)
-    | Just (tCon, ts, field, flv) <- isSatNewTyCon s t = newtypeCoercion tCon ts field flv
-    | Just (tCon, ts, field, flv) <- isSatNewTyCon t s = swap <$> newtypeCoercion tCon ts field flv
-    | SymCo co' <- co = swap <$> convertCoercion env co'
-    | SubCo co' <- co = convertCoercion env co'
-    | otherwise = lift $ unhandled "Coercion" co
+convertCoercion :: Env -> Coercion -> ConvertM (LF.Expr -> LF.Expr, LF.Expr -> LF.Expr)
+convertCoercion env co = evalStateT (go env co) 0
   where
+    go :: Env -> Coercion -> StateT Int ConvertM (LF.Expr -> LF.Expr, LF.Expr -> LF.Expr)
```

Comment:
This diff is mostly due to an additional level of indentation. "Hide whitespace changes" is your friend here.